app/scripts/models/auth_brokers/pairing/authority.js

```diff
@@ -0,0 +1,110 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Manages the OAuth flow by webchannel messages to the browser,
+// to help with a pairing-based flow.
+
+import AuthorityStateMachine from '../../pairing/authority-state-machine';
+import OAuthAuthenticationBroker from '../oauth';
+import setRemoteMetaData from './remote-metadata';
+
+const PAIR_HEARTBEAT_INTERVAL = 1000;
+
+export default class AuthorityBroker extends OAuthAuthenticationBroker {
+  type = 'authority';
+
+  initialize (options) {
+    super.initialize(options);
+
+    const { notifier } = options;
+
+    // The AuthorityStateMachine is responsible for driving the next steps of the pairing process.
+    // It transitions between various pairing views.
+    this.stateMachine = new AuthorityStateMachine({}, {
+      broker: this,
+      notifier,
+      relier: this.relier
+    });
+  }
+
+  fetch () {
+    return Promise.resolve()
+      .then(() => super.fetch())
+      .then(() => this.getSupplicantMetadata())
+      .then(() => this.startHeartbeat());
+  }
+
+  /**
+   * The heartbeat methods below communicate the 'fxaccounts:pair_heartbeat' message.
+   * See the pairing documentation for the possible formats of the heartbeat message.
```

Comment:
It doesn't look like the "pairing documentation" is part of this PR: should it also be extracted from #6876 and land with this initial piece?

Reply:
I'm not sure

Reply:
I'm just gonna do a follow up PR with just the docs. Please trust me that the "the pairing documentation" will appear :)

Reply:
👍 you have my complete trust 😀